src/main/kotlin/net/maizegenetics/phgv2/pathing/HaploidPathFinding.kt

```diff
@@ -0,0 +1,273 @@
+        .required()
+        .validate { require(File(it).exists()) {"$it is not a valid file"} }
+
+    val hvcfDir by option(help = "The directory containing the hvcf files used to build a HaplotypeGraph for path finding.")
+        .required()
+        .validate { require(File(it).isDirectory) {"$it is not a valid directory."} }
+
+    val referenceGenome by option(help = "path to reference genome (fasta or fastq")
+        .required()
+        .validate { require(File(it).exists()) {"$it is not a valid file"} }
+
+    val outputDir by option(help = "The directory where the output hvcfs will be written. The output file names will be <sampleName>.h.vcf.")
+        .required()
+        .validate { require(File(it).isDirectory) {"$it is not a valid directory."} }
+
+    val probCorrect by option(help = "The probability that a mapped read was mapped correctly")
+        .double()
+        .default(0.99)
+        .validate { require(it in 0.5..1.0) {"prob-correct must be between 0.5 and 1.0"} }
+
+    val probSameGamete by option(help = "The probability of transitioning to the same gamete in the next reference range")
+        .double()
+        .default(0.99)
+
+    val minGametes by option(help = "The minimum number of gametes with a haplotype in a reference range. " +
+            "Reference ranges with fewer gametes will not be imputed.")
+        .int()
+        .default(2)
+        .validate { require(it > -1) {"min-gametes must be a positive integer"} }
+
+    val minReads by option(help = "The minimum number of reads per ReferenceRange. Reference ranges with fewer reads will not be imputed")
+        .int()
+        .default(0)
+        .validate { require(it > -1) {"min-reads must be a positive integer."} }
+
+    val maxReadsPerKb by option(help = "ReferenceRanges with more than max-reads-per-kb will not be imputed.")
+        .int()
+        .default(1000)
+        .validate { require(it > -1) {"max-reads-per-kb must be a positive integer."} }
+
+    val useLikelyAncestors by option(help="Use only the most likely ancestors of each sample for path finding.")
+        .boolean()
+        .default(false)
+
+    val maxAncestors by option(help = "If use-likely-ancestors = true, use at most max-ancestors.")
+        .int()
+        .default(Int.MAX_VALUE)
+
+    val minCoverage by option(help = "If use-likely-ancestors = true, use the fewest number of ancestors that together have this proportion of mappable reads.")
+        .double()
+        .default(1.0)
+        .validate { require(it in 0.5..1.0) {"min-coverage must be between 0.5 and 1.0"} }
+
+    val threads by option(help = "number of threads used to find paths.").int().default(3)
+
+
+    private val myLogger = LogManager.getLogger(HaploidPathFinding::class.java)
+
+    private fun buildHaplotypeGraph(): HaplotypeGraph {
+        val listOfHvcfFilenames = File(hvcfDir).listFiles().filter { it.name.endsWith(".h.vcf") }.map { it.path }
```

Comment:
Should this check for compressed files?  If this is the same hvcf folder used when loading tiledb it is expected that the files are compressed with an index.